protobuf/lib/src/protobuf/pb_map.dart

```diff
@@ -39,6 +39,35 @@ class PbMap<K, V> extends MapBase<K, V> {
     _wrappedMap[key] = value;
   }
 
+  @override
+  bool operator ==(other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other is! PbMap) {
+      return false;
+    }
+    if (other.length != length) {
+      return false;
+    }
+    if (other.hashCode != hashCode) {
```

Comment:
I think calculating the hash code during equality testing is slower than just doing structural equality testing.

Reply:
Much slower if the first key is not in the other map!

It is also exponential in the nesting depth of the PbMaps when the map values contain PbMaps -
you walk the maps for this hashCode call, and again in the `!=` below.

Perhaps you can make a benchmark for PbMaps nested 20x deep to prove we avoid this problem.